lib/worker.js

```diff
@@ -0,0 +1,68 @@
+const {Task} = require('atom');
+const {default: PQueue} = require('p-queue');
+const uniqueString = require('unique-string');
+
+const queue = new PQueue({concurrency: 1});
+let worker = null;
+
+module.exports.startWorker = startWorker;
+module.exports.stopWorker = stopWorker;
+
+function startWorker() {
+	if (worker) {
+		if (worker.childProcess.connected) {
+			return;
+		}
+
+		stopWorker();
+	}
+
+	worker = new Task(require.resolve('./xo-helper.js'));
+
+	worker.start([]);
+}
+
+function stopWorker() {
+	if (worker) {
+		worker.terminate();
+		worker = null;
+	}
+}
+
+function sendJob(config) {
+	config.id = uniqueString();
+
+	let errorDisposable;
+	let responseDisposable;
+
+	return new Promise((resolve, reject) => {
```

Comment:
I think you could use https://github.com/sindresorhus/p-event to simplify the handling here.

Reply:
[`task.on(eventName, callback)`](https://flight-manual.atom.io/api/v1.57.0/Task/#instance-on) returns a [`Disposable`](https://flight-manual.atom.io/api/v1.57.0/Disposable/)

It doesn't look like `p-event` would be able to handle tear down of a disposable?

Reply:
Hmm, right. I missed that.